cmd/snap-confine/snap-confine.rst

```diff
@@ -129,9 +129,13 @@ FILES
 
 	Description of the mount profile.
 
-`/var/lib/snapd/seccomp/bpf/*`:
+`/var/lib/snapd/seccomp/bpf/*.bpf`:
 
-	Description of the seccomp profile.
+	Compiled seccomp profile program.
+
+`/var/lib/snapd/seccomp/bpf/*.in`:
+
+	Input for the /usr/lib/snapd/snap-seccomp profile compiler.
```

Comment:
In a better world we would do exactly what you describe. Just use /var/lib/snapd/seccomp/profiles/snap.$security_tag` as the input and compile to the bpf in a cache dir. However we also need to deal with https://forum.snapcraft.io/t/snapd-2-25-blocked-because-of-revert-race-condition/ - so we cannot write an incompatible new snap.$security_tag because in the revert case we will fail in the same way as outlined in the forum post. The only way around this is a new directory to ensure we keep a compatible seccomp profile for old snapd around.